File: server-net/Paid.Domain/Models/InvoiceDetailModel.cs
using System;
using System.Collections.Generic;
using Paid.Domain.Models;

public class InvoiceDetailModel
{
    public int InvoiceId { get; set; }
    public DateTime InvoiceDate { get; set; }
    public DateTime DueDate { get; set; }
    public ProjectModel Project { get; set; }
    public DateTime PeriodStartDate { get; set; }
    public DateTime PeriodEndDate { get; set; }
    public ICollection<WorkItemModel> WorkItems { get; set; }
}

File: web/src/app/models/InvoiceDetailModel.ts
export interface InvoiceDetailModel {
  invoiceId: number;
  invoiceDate: Date;
  dueDate: string;
  project: ProjectModel;
  periodStartDate: string;
  periodEndDate: string;
  workItems: WorkItemModel[];
}

export interface ProjectModel {
  id: number;
  name: string;
  description: string;
  rateCents: number;
  isActive: boolean;
  addressLine1: string;
  addressLine2: string;
  city: string;
  province: string;
  postalCode: string;
  country: string;
}

export interface WorkItemModel {
  id: number;
  description: string;
  durationMins: number;
}
